tgui/packages/tgui/interfaces/NtosPortraitPrinter.js

```diff
@@ -1,43 +1,83 @@
 import { resolveAsset } from '../assets';
 import { useBackend, useLocalState } from '../backend';
-import { Button, NoticeBox, Section, Stack } from '../components';
+import { Button, NoticeBox, Section, Stack, Input } from '../components';
 import { NtosWindow } from '../layouts';
 
 export const NtosPortraitPrinter = (props, context) => {
   const { act, data } = useBackend(context);
   const [listIndex, setListIndex] = useLocalState(context, 'listIndex', 0);
   const {
     paintings,
+    search_string,
+    search_mode,
   } = data;
-  const current_portrait_title = paintings[listIndex]["title"];
-  const current_portrait_asset_name = "paintings" + "_" + paintings[listIndex]["md5"];
+  const got_paintings = !!paintings.length;
+  const current_portrait_title = got_paintings
+    && paintings[listIndex]["title"];
+  const current_portrait_author = got_paintings
+    && "By " + paintings[listIndex]["creator"];
+  const current_portrait_asset_name = got_paintings
+    && "paintings" + "_" + paintings[listIndex]["md5"];
   return (
     <NtosWindow
       title="Art Galaxy"
       width={400}
-      height={406}>
+      height={446}>
       <NtosWindow.Content>
         <Stack vertical fill>
+          <Stack.Item>
+            <Section
+              title="Search">
+              <Input fluid
+                placeholder="Search Paintings..."
+                value={search_string}
+                onChange={(e, value) => {
+                  act('search', {
+                    to_search: value,
+                  });
```

Comment:
Wouldn't you want this to be done only when they press enter, to avoid too much querying for intermediary steps? Would probably save a decent bit on performance on servers with a larger database of paintings.

Reply:
You're confusing `onChange` with `OnInput`. This only fires when the text bar is unfocused or enter is pressed.